src/models/release-asset.ts

```diff
@@ -0,0 +1,468 @@
+import type {
+	ODataOptionsWithoutCount,
+	OptionsToResponse,
+} from 'pinejs-client-core';
+import type {
+	InjectedDependenciesParam,
+	InjectedOptionsParam,
+	Release,
+	ReleaseAsset,
+} from '..';
+import { mergePineOptions } from '../util';
+import type { ReleaseRawVersionApplicationPair } from './release';
+import pLimit from 'p-limit';
+import { getType } from 'mime';
+
+const getMimeType = (filePath: string): string => {
+	return getType(filePath) ?? 'application/octet-stream';
+};
+
+const getFileBasename = (filePath: string): string => {
+	return filePath.split('/').pop() ?? filePath.split('\\').pop() ?? filePath;
+};
+
+type WriteReleaseAssetParams = Omit<ReleaseAsset['Write'], 'asset'> & {
+	asset?: File | string | null;
+};
+type UploadPart = {
+	url: string;
+	chunkSize: number;
+	partNumber: number;
+};
+
+type CommitData = {
+	uuid: string;
+	providerCommitData: {
+		Parts: Array<{ PartNumber: number; ETag: string }>;
+	};
+};
+
+type ReleaseAssetBeginUpload = {
+	asset: {
+		uuid: string;
+		uploadParts: UploadPart[];
+	};
+};
+
+type UploadParams = {
+	chunkSize?: number | undefined;
+	parallelUploads?: number | undefined;
+};
+
+type ReleaseAssetId =
+	| ReleaseAsset['Read']['id']
+	| {
+			asset_key: ReleaseAsset['Read']['asset_key'];
+			release: string | number | ReleaseRawVersionApplicationPair;
+	  };
+
+const MINIMUM_MULTIPART_SIZE = 5 * 1024 * 1024; // 5MB
+
+const DEFAULT_MULTIPART_CHUNK_SIZE = 5 * 1024 * 1024; // 5MB
```

Comment:
Heads-up that these atm are `MiB` and for MB you could just do
```suggestion
const MINIMUM_MULTIPART_SIZE = 5_000_000; // 5MB

const DEFAULT_MULTIPART_CHUNK_SIZE = 5_000_000; // 5MB
```
No sure which one of the two we intend to be using

Reply:
Yeah, 5MiB is what we want (altough the naming with the `i` is new for me)